UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs

```diff
@@ -56,11 +60,18 @@ public bool TryAddSheet(ItemTrait material, int quantity)
 
 		public int TryRemoveSheet(ItemTrait material, int quantity)
 		{
+			var Cm3Used = Cm3PerSheet * quantity;
+			foreach (var materialToTake in MaterialList)
+			{
+				if(materialToTake.Key != material) continue;
+				if(materialToTake.Value > Cm3Used || materialToTake.Value == 0) continue;
+				ConsumeMaterial(material, materialToTake.Value);
+				return quantity;
+			}
 			if (MaterialList[material] < Cm3PerSheet)
 			{
 				return 0;
 			}
-			var Cm3Used = Cm3PerSheet * quantity;
 			if (MaterialList[material] < Cm3Used)
 			{
 				quantity = MaterialList[material] / Cm3PerSheet;
```

Comment:
Seems a little complex, and you're not poking the UI.

Wouldn't it be simpler instead to just remove the `if` condition on line `63` (before changes) and get the min. of the division result and the requested amount?
```suggestion
			quantity = Mathf.Min(quantity, MaterialList[material] / Cm3PerSheet);
```
Can't suggest changes on lines too far away so there's an exta `}`... @Github...

Reply:
I don't understand what you want here?

Reply:
Just do 

```
		public int TryRemoveSheet(ItemTrait material, int quantity)
		{
			quantity = Mathf.Min(quantity, MaterialList[material] / Cm3PerSheet);
			var Cm3Used = Cm3PerSheet * quantity;
			ConsumeMaterial(material, Cm3Used);
			UpdateGUIs.Invoke();
			return quantity;
		}
```

Reply:
Done and tested